### PR TITLE
Adding Coremark example.

### PR DESCRIPTION
**Pull request by Jacob-Scheiffler**
The Coremark example was originally written for the MAX32655. This PR adds the original example along and a ported version of the example to each of the other micros.

### CODE REVIEW

**Comment by sihyung-maxim** [comment]
The icc_regs.h was removed from the ME55. Just found a new bug in the SVD checker though. An unsupported register file was removed, and the SVD action doesn't check for register file removals and sees the "removal" as a register file change. Because the register file isn't supported according to the list of supported peripherals per part in the SVD scripts and the SVD action sees this as a register file change, the SVD action will attempt to generate icc_regs.h but fails when it can't find the unsupported register file's corresponding peripheral SVD file. 

I will have to update this once this PR is merged because the Verify_Register_SVD workflow has had some updates.

**Review by sihyung-maxim — APPROVED** [review]
Interesting how the main program is located in Libraries/Coremark/core_main.c, but the examples only contain core_portme.c/.h.

File: Examples/MAX32520/Coremark/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+## Description
+
+This example determines the Coremark rating for the MAX32xxx and MAX78xxx series microcontrollers.
```
Comment (Jake-Carter) [inline Examples/MAX32520/Coremark/README.md:3]: Needs more info on what Coremark is and what the example tells you.

File: Examples/MAX32520/Coremark/project.mk [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# "Makefile" that is located next to this one.
+
+# For instructions on how to use this system, see
+# https://github.com/Analog-Devices-MSDK/VSCode-Maxim/tree/develop#build-configuration
+
+# **********************************************************
+
+# Add your config here!
+
+IPATH+=$(MAXIM_PATH)/Libraries/Coremark
```
Comment (Jake-Carter) [inline Examples/MAX32520/Coremark/project.mk:12]: Needs comments documenting the build setup

File: Examples/MAX32520/Coremark/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
```
Comment (Jake-Carter) [inline Examples/MAX32520/Coremark/README.md:4]: Please add the following contents to the README, which have now become a part of the new [standard readme template in MSDKGen](https://github.com/Analog-Devices-MSDK/MSDKGen/tree/main/Markdown/Template).

```Markdown
### Project Usage

Universal instructions on building, flashing, and debugging this project can be found in the **[MSDK User Guide](https://analog-devices-msdk.github.io/msdk/USERGUIDE/)**.

### Project-Specific Build Notes

(None - this project builds as a standard example)
```

This project doesn't offer any build config options, but you may want to expand a bit on exactly where the source code is pulled from in the build notes section.

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Yeah please add some comments to project.mk and some documentation in the readme to clarify what the Coremark library is and how these projects are built.

**Comment by Jake-Carter** [comment]
@sihyung-maxim SVD verification is failing for this PR?

**Comment by Jacob-Scheiffler** [comment]
> Yeah please add some comments to project.mk and some documentation in the readme to clarify what the Coremark library is and how these projects are built.

@Jake-Carter pretty sure I made all the changes you asked for, mind giving it another look?

**Review by Jake-Carter — APPROVED** [review]
Looks good!

**Pull request merged**